crepecake/src/main/java/com/google/cloud/tools/crepecake/blob/BlobStream.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.blob;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.security.DigestException;
+import java.security.NoSuchAlgorithmException;
+import javax.annotation.Nonnull;
+
+/** A stream for BLOBs. */
+public interface BlobStream {
+
+  /**
+   * Writes the BLOB to an {@link OutputStream}.
+   *
+   * @param outputStream the {@link OutputStream} to write to
+   */
+  void writeTo(OutputStream outputStream)
```

Comment:
It's unclear why this general interface should throw NoSuchAlgorithmException or DigestException. I'm not sure it does throw NoSuchAlgorithmException. If DigestException is needed, then an @param is called for to explain it.

Reply:
The NoSuchAlgorithmException is because the writing is mostly done through CountingDigestOutputStream, which performs a SHA-256 hash as the output is written. I wrapped this in IOException to make this interface more clear. 

DigestException seems to still fit this interface since getWrittenBlobDescriptor is getting the digest. I added an explanation for DigestException.